src/pages/ExamsPage/components/AttemptList.jsx

```diff
@@ -1,14 +1,15 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Button, DataTable } from '@edx/paragon';
+import { DataTable } from '@edx/paragon';
 import { useIntl } from '@edx/frontend-platform/i18n';
+import ResetExamAttemptButton from './ResetExamAttemptButton';
 
 const capitalizeFirstLetter = (string) => string.charAt(0).toUpperCase() + string.slice(1);
 
+// This component has to be compartmentalized here
+// Otherwise we will break React rules.
```

Comment:
can you give a bit more detail on the rule and why we need this?

Reply:
Hi! I was kind of vague on this, so to specify, if we pass in the `ResetExamAttemptButton` directly into the "Cell" value for the "Action" column as such:
```
      <DataTable
        isPaginated
        initialState={{
          pageSize: 20,
        }}
        isSortable
        itemCount={attempts.length}
        additionalColumns={[
          {
            id: 'action',
            Header: formatMessage({
              id: 'AttemptsList.action',
              defaultMessage: 'Action',
              description: 'Table header for the table column listing action to reset the exam attempt',
            }),
            Cell: ({ row }) => <ResetExamAttemptButton row={row} />
            ,
          },
        ]}
```
we get this lint error:
```
Do not define components during render. React will see a new component type on every render and destroy the entire subtree’s DOM nodes and state (https://reactjs.org/docs/reconciliation.html#elements-of-different-types). Instead, move this component definition out of the parent component “AttemptList” and pass data as props. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
```
I _believe_ this means that not compartmentalizing the child component makes things far less efficient. Let me know if you think this comment is necessary, or maybe if we should ignore this rule altogether.

Reply:
I got no idea on this one, it could be worth asking the frontend group about to get more insight. I would update the comment to mention the unstable-nested-component problem specifically. Otherwise it's unclear to the reader what the problem is.